### PR TITLE
SAK-32104 - Sitestats - Lessons: number of pages is incorrectly calculated

### PR DESCRIPTION
**Pull request by mpellicer**
@adrianfish Would you mind to take a look?

I admit this solution is very radical but correct, use the Lessons API instead of events, events are incorrect in duplicated courses because there are no "create" events.

### CODE REVIEW

**Comment by jonespm** [comment]
Yeah, this looks fine. The getTotalResources essentially does the same because countFilesUsingCHS is always true running except during the unit tests. 

We probably should have some event listened for and the stats updated when a site is duplicated, but this is a lot more effort.

I might change this method, leaving this query in here and protecting it the same was as getTotalResources with countLEssonsUsingLBS = true. (and a setter in there)

```
			if(countLessonsUsingLBS) {
// Your new code
 			}else{
// All this SQL
```

Then it will be able to work in unit tests (if they are ever written) AND if the underlying issue on duplication is ever fixed we could switch this around. Then it's a little less radical.

**Comment by jonespm** [comment]
I also believe that getTotalResources would have this same issue if countResourcesUsingCHS was set to false. This was a "performance" fix added back in 2009. But because of these edge cases was probably disabled by default in https://jira.sakaiproject.org/browse/SAK-28799.

I think it's a good idea that would still work if this was fixed.
"Unfortunately, this leads to wrong results if site had resources activity before adding SiteStats tool "

Basically if the site stats is added to an existing site it would need to recalculate the statistics, a feature never implemented. ;)

**Comment by mpellicer** [comment]
@jonespm 

Thanks for taking a look, I've updated the PR with the requested changes.

I'll update the documentation here after the merge:
https://confluence.sakaiproject.org/display/STAT/Configuration+Options

**Comment by jonespm** [comment]
Looks great, will merge after travis finishes. Thanks.
